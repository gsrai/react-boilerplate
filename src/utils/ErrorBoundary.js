import React, { Component } from 'react'
import PropTypes from 'prop-types'

class ErrorBoundary extends Component {
  constructor(props) {
    super(props)
    this.state = { hasError: false }
  }

  componentDidCatch(error, information) {
    this.setState({ hasError: true })
    console.error(`ErrorBoundary: ${error} ${information} Error`)
  }

  render() {
    const { children } = this.props
    const { hasError } = this.state
    if (hasError) {
      return <h1>Something went wrong.</h1>
    }
    return children
  }
}

ErrorBoundary.propTypes = {
  children: PropTypes.node.isRequired
}

export default ErrorBoundary
