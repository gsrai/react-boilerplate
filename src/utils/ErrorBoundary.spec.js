import { shallow, mount } from 'enzyme'
import React from 'react'
import ErrorBoundary from './ErrorBoundary'

const errMsg = 'Oops something went wrong!'
const throwSomething = _ => { throw new Error(errMsg) }

class ErrorComp extends React.Component {
  render() {
    return (
      <React.Fragment>
        {throwSomething()}
      </React.Fragment>
    )
  }
}

describe('Component Rendering', () => {
  test('component should render correctly', () => {
    const wrapper = shallow(
      <ErrorBoundary>
        <div>foo</div>
      </ErrorBoundary>
    )
    expect(wrapper).toMatchSnapshot()
  })

  test('component should render correctly when error', () => {
    const wrapper = mount(
      <ErrorBoundary>
        <ErrorComp />
      </ErrorBoundary>
    )
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.state().hasError).toBe(true)
  })
})
