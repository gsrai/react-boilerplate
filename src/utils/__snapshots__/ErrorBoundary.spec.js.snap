// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Rendering component should render correctly 1`] = `
<div>
  foo
</div>
`;

exports[`Component Rendering component should render correctly when error 1`] = `
<ErrorBoundary>
  <h1>
    Something went wrong.
  </h1>
</ErrorBoundary>
`;
